packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx

```diff
@@ -99,9 +124,108 @@ const CompactInteractiveList = (props) => {
     width,
     columnMinimumWidth,
     columnMaximumWidth,
+    onCellSelect,
+    onClearSelection,
   } = props;
 
   const theme = useContext(ThemeContext);
+  const listRef = React.useRef();
+
+  // using ref so that keyboard navigation and focusing on cells won't trigger component re-render.
+  const focusedCell = useRef({ row: 0, cell: 0 });
+
+  const setFocusOnCell = ({ row, cell }) => {
+    // add 1 to the row number to accomodate for hidden header
+    const focusedCellElement = listRef.current.children[row + 1].children[cell];
+    const interactiveChildren = getFocusableElements(focusedCellElement);
+    if (interactiveChildren?.length > 0 && interactiveChildren[0]) {
+      // currently a cell can have only one interact-able element in it, which gets auto-focused.
+      interactiveChildren[0].focus();
+    } else {
+      // cell gets focus if there is no interact-able elements in it.
+      focusedCellElement.focus();
+    }
+  };
+
+  const setFocusedRowCol = ({ row, cell }) => {
+    focusedCell.current = { row, cell };
+    setFocusOnCell({ row, cell });
+  };
+
+  const handleOnCellSelect = useCallback(({ row, cell }) => {
+    focusedCell.current = { row, cell };
+    // check for interactive elements in that cell
+    // add 1 to the row number to accomodate for hidden header
+    const elementToFocus = listRef.current.children[row + 1].children[cell];
+    const interactiveChildren = getFocusableElements(elementToFocus);
+    // if no interactive elements
+    if (onCellSelect && interactiveChildren?.length === 0) {
+      onCellSelect(rows[row].id, columns[cell].id);
+    }
+  }, [onCellSelect, rows, columns]);
+
+  const handleKeyDown = (event) => {
+    let moveFocusTo = focusedCell.current;
```

Comment:
Maybe we could rename this variable to something like, nextCell, or something similar.  I think it would be helpful to specify that this is a cell.  The current name looks like a method.

Reply:
I was trying to be consistent with the same method for similar functionality in DataGrid, thought it would be less confusing for developers if the naming is the same: https://github.com/cerner/terra-framework/blob/ba1d238e55bfa51b74723029baa89b34f6dea65d/packages/terra-data-grid/src/DataGrid.jsx#L374

Reply:
I don't see a variable named, "moveFocusTo", in that file.  Maybe I am not understanding properly.